playbooks/init/main.yml

```diff
@@ -12,6 +13,12 @@
         installer_phase_initialize:
           status: "In Progress"
           start: "{{ lookup('pipe', 'date +%Y%m%d%H%M%SZ') }}"
+  - name: Ensure Ansible facts directory exists
+    file:
+      mode: 0777
```

Comment:
Can you try 755 which appears to be the default permissions that /etc/ansible and /etc/ansible/facts.d have on my hosts?

Reply:
I'll try it...

Reply:
Error log from using mod 755...
```
TASK [Gather Cluster facts and set is_containerized if needed] **********************************************************************
fatal: [ec2-52-90-109-175.compute-1.amazonaws.com]: FAILED! => {
    "changed": false,
    "failed": true,
    "module_stderr": "Shared connection to ec2-52-90-109-175.compute-1.amazonaws.com closed.\r\n",
    "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_zeQ28r/ansible_module_openshift_facts.py\", line 1861, in <module>\r\n    main()\r\n  File \"/tmp/ansible_zeQ28r/ansible_module_openshift_facts.py\", line 1848, in main\r\n    additive_facts_to_overwrite)\r\n  File \"/tmp/ansible_zeQ28r/ansible_module_openshift_facts.py\", line 1507, in __init__\r\n    additive_facts_to_overwrite)\r\n  File \"/tmp/ansible_zeQ28r/ansible_module_openshift_facts.py\", line 1523, in generate_facts\r\n    additive_facts_to_overwrite)\r\n  File \"/tmp/ansible_zeQ28r/ansible_module_openshift_facts.py\", line 1727, in init_local_facts\r\n    save_local_facts(self.filename, new_local_facts)\r\n  File \"/tmp/ansible_zeQ28r/ansible_module_openshift_facts.py\", line 1175, in save_local_facts\r\n    \"Could not create fact file: %s, error: %s\" % (filename, ex)\r\n__main__.OpenShiftFactsFileWriteError: Could not create fact file: /etc/ansible/facts.d/openshift.fact, error: [Errno 13] Permission denied: '/etc/ansible/facts.d/openshift.fact'\r\n",
    "rc": 0
}

MSG:

MODULE FAILURE

	to retry, use: --limit @/home/ccallega/git/openshift-ansible/playbooks/aws/openshift-cluster/build_ami.retry
```

Reply:
I can make 755 work with this additional diff...
```
diff --git a/playbooks/init/facts.yml b/playbooks/init/facts.yml
index d41f365dc..b173080ee 100644
--- a/playbooks/init/facts.yml
+++ b/playbooks/init/facts.yml
@@ -6,6 +6,7 @@

 - name: Initialize host facts
   hosts: oo_all_hosts
+  become: true
   tasks:
   - name: load openshift_facts module
     import_role:
```

Reply:
@sdodson  Looks like we have two choices here ...
1) mod 777 on /etc/ansible/facts.d
2) mod 755 and Task 'Initialize host facts' with 'become: true'

I'm good with either/or...up to you.

Reply:
We can't 777 mod /etc/ansible/facts.d, that would be a security issue.

become: true is typically set by the user when running plays.  If you're not using the root user to configure hosts with ansible, then you must use become.  All of our plays and roles assume privileged access.